File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> stack;
        for(int i=0;i<s.length();i++){
            if(s[i]=='('||s[i]=='['||s[i]=='{'){
                stack.push(s[i]);
            }
            else{
                if(stack.empty())
                    return false;
                char tmp=stack.top();
            
                if(s[i]==')' && tmp!='(' || s[i]==']' && tmp!='[' ||
                  s[i]=='}' && tmp!='{'){
                    return false;
                }
                stack.pop();
            }
            }
        if(stack.empty())
            return true;
        else
            return false;
        }
};

File: 0020-valid-parentheses/0020-valid-parentheses.cs
public class Solution {
    public bool IsValid(string s) {
        Stack<char> stack=new();
        for(int i=0;i<s.Length;i++){
            if(s[i]=='('||s[i]=='{'||s[i]=='[')
                stack.Push(s[i]);
            else{
                if(stack.Count==0) return false;
                if(s[i]==')'){
                    if(stack.Peek()=='(')
                        stack.Pop();
                    else
                        return false;
                }
                else if(s[i]=='}'){
                    if(stack.Peek()=='{')
                        stack.Pop();
                    else
                        return false;
                }
                else if(s[i]==']'){
                    if(stack.Peek()=='[')
                        stack.Pop();
                    else
                        return false;
                }
            }
        }
        return stack.Count()==0;
    }
}